syntax = "proto3";

message LTLSatProblem {
    string formula = 1;
    bool simplify = 2;
    float timeout = 3;
}

message LTLSatSolution {
    string status = 1;
    string trace = 2;
}

message LTLTraceMCProblem {
    string formula = 1;
    string trace = 3;
    string timeout = 4;
}

message TraceMCSolution {
    string status = 1;
}

message LTLSpecification {
    repeated string inputs = 1;
    repeated string outputs = 2;
    repeated string guarantees = 3;
    repeated string assumptions = 4;
}